src/types/nbodysystem.rs

```diff
@@ -98,4 +123,106 @@ impl NBodySystem {
                 },
             )
     }
+
+    pub fn set_output(&mut self, output: String) {
+        self.m_output = output;
+    }
+
+    pub fn set_limit(&mut self, limit: u16) {
+        self.m_limit = limit;
+    }
+
+    pub fn set_file(&mut self, file: bool) {
```

Comment:
From the architectural side, the simulation system itself shouldn't care about what's the state observer(is it GUI or CLI or a file). So, technically it's better to consider usage of Observer pattern there with introducing the SimulationState primitive which allows to notify external observers about the simulation world change.

Reply:
Done, implemented the signal approach